app/assets/javascripts/mumuki_laboratory/application/console.js

```diff
@@ -36,65 +35,64 @@ var mumuki = mumuki || {};
     $('.jquery-console-prompt-box:not(:last)').remove()
   }
 
-  function QueryConsole() {
-    this.endpoint = $('#console_endpoint').val();
-    this.token = new mumuki.CsrfToken();
-    this.statefulConsole = $('#stateful_console').val() === "true";
-    this.lines = [];
-  }
-
-  QueryConsole.prototype = {
-    newQuery: function (line) {
+  class QueryConsole {
+    constructor() {
+      this.endpoint = $('#console_endpoint').val();
+      this.token = new mumuki.CsrfToken();
+      this.statefulConsole = $('#stateful_console').val() === "true";
+      this.lines = [];
+      this.controller = null;
```

Comment:
is ths necessary? :thinking:

Reply:
Not really, but I added to make IDE aware of the existence of this field. And to better describe this class state.